### PR TITLE
Hello Paul Fixed The Module

### PR DESCRIPTION
**Pull request by timnboys**
Hello Paul Fixed The Module

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "blesta/digitaloceanmodule",
+    "description": "DigitalOcean",
+    "license": "proprietary",
```
Comment (pmphillips) [inline composer.json]: License should be "MIT"

File: digitaloceanmodule.php [hunk 2]
```diff
@@ -223,6 +223,7 @@ public function getImagesDropdown($p_id) {
         }
         return $array;
     }
+
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Unnecessary new line should be removed

File: apis/digitalocean_api.php [hunk 3]
```diff
@@ -50,12 +50,13 @@ private function makeGetConnection($action) {
...
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+			'Authorization: Bearer ' . $this->apiKey
         ));
         curl_setopt($ch, CURLOPT_URL, $action);
         curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_USERPWD, $this->apiKey . ':' . "");
+        //curl_setopt($ch, CURLOPT_USERPWD, $this->apiKey . ':' . "");
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: Should be removed if not needed, rather than commented out.

File: apis/digitalocean_api.php [hunk 4]
```diff
@@ -71,11 +72,12 @@ private function makePostConnection($action, array $post_data = NULL) {
...
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+			'Authorization: Bearer ' . $this->apiKey
         ));
         curl_setopt($ch, CURLOPT_POST, count($result_p));
         curl_setopt($ch, CURLOPT_POSTFIELDS, $result_p);
-        curl_setopt($ch, CURLOPT_USERPWD, $this->apiKey . ':' . "");
+       //curl_setopt($ch, CURLOPT_USERPWD, $this->apiKey . ':' . "");
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: Should be removed if not needed, rather than commented out.

File: composer.json (hunk 1 above)
Comment (timnboys) [reply composer.json]: I just had that from a while ago sorry that was from the form post I will dump that and put MIT on it then.

**Comment by timnboys** [comment]
okay paul I just fixed all issues as you said.

File: apis/digitalocean_api.php [hunk 5]
```diff
@@ -50,12 +50,12 @@ private function makeGetConnection($action) {
         $ch = curl_init();
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+			'Authorization: Bearer ' . $this->apiKey
```
Comment (tysonphillips) [inline apis/digitalocean_api.php]: Can you fix the tabbing here? Each array element on its own line should be vertically aligned by the same indentation.

File: apis/digitalocean_api.php [hunk 6]
```diff
@@ -71,11 +71,11 @@ private function makePostConnection($action, array $post_data = NULL) {
         curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE);
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+			'Authorization: Bearer ' . $this->apiKey
```
Comment (tysonphillips) [inline apis/digitalocean_api.php]: Can you fix the tabbing here? Each array element on its own line should be vertically aligned by the same indentation.

File: apis/digitalocean_api.php [hunk 7]
```diff
@@ -87,13 +87,13 @@ private function makeDeleteConnection($action) {
         $ch = curl_init();
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+			'Authorization: Bearer ' . $this->apiKey
```
Comment (tysonphillips) [inline apis/digitalocean_api.php]: Can you fix the tabbing here? Each array element on its own line should be vertically aligned by the same indentation.

File: apis/digitalocean_api.php [hunk 8]
```diff
@@ -104,15 +104,15 @@ private function makePutConnection($action, array $put_data = NULL) {
...
```
Comment (tysonphillips) [inline apis/digitalocean_api.php]: Can you fix the tabbing here? Each array element on its own line should be vertically aligned by the same indentation.

File: apis/digitalocean_api.php (hunk 6 above)
Comment (tysonphillips) [inline apis/digitalocean_api.php]: Can you fix the tabbing here? Each array element on its own line should be vertically aligned by the same indentation.

**Comment by timnboys** [comment]
fixed tyson.

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "blesta/digitaloceanmodule",
```
Comment (tysonphillips) [inline composer.json:2]: You should include yourself as an author. See https://getcomposer.org/doc/04-schema.md#authors

File: digitaloceanmodule.php [hunk 10]
```diff
@@ -3,7 +3,7 @@
 class Digitaloceanmodule extends Module {
 
     private static $version = "2.2";
-    private static $authors = array(array('name' => "ModulesBakery.com.", 'url' => "http://www.modulesbakery.com"));
+    private static $authors = array(array('name' => "ModulesBakery.com.", 'url' => "http://www.modulesbakery.com"),array('name' => "CubeData", 'url' => "https://cubedata.net"));
```
Comment (tysonphillips) [inline digitaloceanmodule.php]: Can you add a space between the current author array and your array? That, or break up the array to limit line length, e.g.

```
private static $authors = array(
    array(
        'name' => ...
        'url' => ...
    ),
    array(
        'name' => ...
        'url' => ...
    )
);
```

File: digitaloceanmodule.php [hunk 11]
```diff
@@ -331,6 +331,30 @@ public function getPackageFields($vars = null) {
         return $fields;
     }
 
+	public function getallImagesDropdown($package) {
```
Comment (tysonphillips) [inline digitaloceanmodule.php]: Method names are expected to be camel-case, so "All" would be capitalized in the name.

File: digitaloceanmodule.php [hunk 12]
```diff
@@ -331,6 +331,30 @@ public function getPackageFields($vars = null) {
...
+        $row = $this->getModuleRow($package->module_row);
+        $api = $this->getApi($row->meta->apiKey);
+        $result = $api->getlongGetResults("images");
+        $dp = array();
+		foreach ($result->images as $key => $value)
+		{
```
Comment (tysonphillips) [inline digitaloceanmodule.php]: The rest of the class places braces on the same line as the condition. You should follow the same syntax.

File: digitaloceanmodule.php [hunk 13]
```diff
@@ -331,6 +331,30 @@ public function getPackageFields($vars = null) {
...
+		$dp[$result->images[$key]->slug] = $result->images[$key]->slug;
+		}
+		return $dp;
+    }
+	
+	public function getAllRegionsDropdown($package) {
+        $row = $this->getModuleRow($package->module_row);
```
Comment (tysonphillips) [inline digitaloceanmodule.php:405]: Indentation looks off here. You should follow the same formatting syntax as the rest of the file.

Comment (timnboys) [reply digitaloceanmodule.php:405]: fixed again.

File: composer.json (hunk 9 above)
Comment (timnboys) [reply composer.json:2]: that has been done.

**Comment by timnboys** [comment]
all of the issues tyson you pointed out has been fixed. @tysonphillips @pmphillips

File: apis/digitalocean_api.php [hunk 14]
```diff
@@ -50,12 +50,12 @@ private function makeGetConnection($action) {
         $ch = curl_init();
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+	    'Authorization: Bearer ' . $this->apiKey
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: This doesn't line up with the array elements above it.

File: apis/digitalocean_api.php [hunk 15]
```diff
@@ -71,11 +71,11 @@ private function makePostConnection($action, array $post_data = NULL) {
         curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE);
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+	    'Authorization: Bearer ' . $this->apiKey
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: This doesn't line up with the array elements above it.

File: apis/digitalocean_api.php [hunk 16]
```diff
@@ -87,13 +87,13 @@ private function makeDeleteConnection($action) {
         $ch = curl_init();
         curl_setopt($ch, CURLOPT_HTTPHEADER, array(
             'Content-Type: application/json',
-            'Accept: application/json'
+            'Accept: application/json',
+	    'Authorization: Bearer ' . $this->apiKey
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: This doesn't line up with the array elements above it.

File: apis/digitalocean_api.php [hunk 17]
```diff
@@ -104,15 +104,15 @@ private function makePutConnection($action, array $put_data = NULL) {
...
```
Comment (pmphillips) [inline apis/digitalocean_api.php]: This doesn't line up with the array elements above it.

File: digitaloceanmodule.php [hunk 18]
```diff
@@ -3,7 +3,14 @@
 class Digitaloceanmodule extends Module {
 
     private static $version = "2.2";
-    private static $authors = array(array('name' => "ModulesBakery.com.", 'url' => "http://www.modulesbakery.com"));
+    private static $authors = array(
+    	array(
+    	'name' => "ModulesBakery.com.",
+    	'url' => "http://www.modulesbakery.com"),
```
Comment (pmphillips) [inline digitaloceanmodule.php]: These array elements should all be tabbed over further.

File: digitaloceanmodule.php [hunk 19]
```diff
@@ -3,7 +3,14 @@
...
+    	array(
+    	'name' => "CubeData", 
+    	'url' => "https://cubedata.net")
```
Comment (pmphillips) [inline digitaloceanmodule.php]: These array elements should all be tabbed over further.

File: digitaloceanmodule.php [hunk 20]
```diff
@@ -222,7 +229,7 @@ public function getImagesDropdown($p_id) {
             }
         }
         return $array;
-    }
+    }
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Unnecessary whitespace after the brace }

File: digitaloceanmodule.php [hunk 21]
```diff
@@ -331,6 +338,28 @@ public function getPackageFields($vars = null) {
         return $fields;
     }
 
+	public function getAllImagesDropdown($package) {
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing for this entire method.

File: digitaloceanmodule.php [hunk 22]
```diff
@@ -331,6 +338,28 @@ public function getPackageFields($vars = null) {
...
+        $api = $this->getApi($row->meta->apiKey);
+        $result = $api->getlongGetResults("images");
+        $dp = array();
+	foreach ($result->images as $key => $value) {
+	     $dp[$result->images[$key]->slug] = $result->images[$key]->slug;
+		}
+	 return $dp;
+    }
+	
+	public function getAllRegionsDropdown($package) {
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing for this entire method.

File: digitaloceanmodule.php [hunk 23]
```diff
@@ -505,7 +534,16 @@ public function getAdminAddFields($package, $vars = null) {
         $domain = $fields->label(Language::_("Digitaloceanmodule.hostname", true), "hostname");
         $domain->attach($fields->fieldText("hostname", $this->Html->ifSet($vars->hostname, $this->Html->ifSet($vars->hostname)), array('id' => "hostname")));
         $fields->setField($domain);
-
+		
+		$region = $fields->label(Language::_("Digitaloceanmodule.region", true), "region");
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 24]
```diff
@@ -505,7 +534,16 @@ public function getAdminAddFields($package, $vars = null) {
...
+        $region->attach($fields->fieldSelect("region", $this->getAllRegionsDropdown($package)), array('id' => "region"));
+        $fields->setField($region);
+		
+		$image = $fields->label(Language::_("Digitaloceanmodule.image", true), "image");
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 25]
```diff
@@ -505,7 +534,16 @@ public function getAdminAddFields($package, $vars = null) {
...
+        $image->attach($fields->fieldSelect("image", $this->getallImagesDropdown($package)), array('id' => "image"));
+        $fields->setField($image);
+		
+		$client_sshkey = $fields->label(Language::_("Digitaloceanmodule.client_sshkey", true), "client_sshkey");
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 26]
```diff
@@ -519,7 +557,15 @@ public function getClientAddFields($package, $vars = null) {
         $domain = $fields->label(Language::_("Digitaloceanmodule.hostname", true), "hostname");
         $domain->attach($fields->fieldText("hostname", $this->Html->ifSet($vars->hostname, $this->Html->ifSet($vars->hostname)), array('id' => "hostname")));
         $fields->setField($domain);
-
+		
+		$region = $fields->label(Language::_("Digitaloceanmodule.region", true), "region");
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 27]
```diff
@@ -519,7 +557,15 @@ public function getClientAddFields($package, $vars = null) {
...
+        $region->attach($fields->fieldSelect("region", $this->getAllRegionsDropdown($package)), array('id' => "region"));
+        $fields->setField($region);
+		
+		$image = $fields->label(Language::_("Digitaloceanmodule.image", true), "image");
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 28]
```diff
@@ -586,14 +632,16 @@ public function addService($package, array $vars = null, $parent_package = null,
             $sshkey_result = $api->getPostResults("account/keys", $ssh_key);
             if (isset($sshkey_result->ssh_key->id) && isset($sshkey_result->ssh_key->name) && $sshkey_result->ssh_key->name === $ssh_key['name']) {
                 $this->log("Create Client SSH Key {$sshkey_result->ssh_key->name}", serialize("sshkey_create"), "input", true);
-                $vars['client_sshkey'] = $sshkey_result->ssh_key->id;
+                $vars['client_sshkey'] = $sshkey_result->ssh_key->fingerprint;
+				$this->log("Set client_sshkey variable", print_r($vars['client_sshkey'],true), "input", true);
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 29]
```diff
@@ -586,14 +632,16 @@ public function addService($package, array $vars = null, $parent_package = null,
...
                 $params = $this->getFieldsFromInput((array) $vars, $package);
                 $result = $api->getPostResults("droplets", $params);
+				$this->log("API CREATE RESULTS" . print_r($result,true), print_r($result,true),"input",true);
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php [hunk 30]
```diff
@@ -649,24 +697,24 @@ public function addService($package, array $vars = null, $parent_package = null,
                 'value' => isset($params['user_data']) ? $params['user_data'] : null,
                 'encrypted' => 0
             ),
-            array(
+			array(
```
Comment (pmphillips) [inline digitaloceanmodule.php]: Please check tabbing.

File: digitaloceanmodule.php (hunk 20 above)
Comment (timnboys) [reply digitaloceanmodule.php]: already done paul.

File: digitaloceanmodule.php (hunk 30 above)
Comment (timnboys) [reply digitaloceanmodule.php]: fixed paul.

File: digitaloceanmodule.php (hunk 29 above)
Comment (timnboys) [reply digitaloceanmodule.php]: fixed.

File: digitaloceanmodule.php (hunk 28 above)
Comment (timnboys) [reply digitaloceanmodule.php]: fixed.

File: digitaloceanmodule.php (hunk 21 above)
Comment (timnboys) [reply digitaloceanmodule.php]: already done paul.

File: digitaloceanmodule.php (hunk 23 above)
Comment (timnboys) [reply digitaloceanmodule.php]: already done paul.

File: digitaloceanmodule.php (hunk 25 above)
Comment (timnboys) [reply digitaloceanmodule.php]: already done paul.

File: digitaloceanmodule.php (hunk 18 above)
Comment (timnboys) [reply digitaloceanmodule.php]: paul that was dumped I made it use the composer way instead.

File: digitaloceanmodule.php [hunk 31]
```diff
@@ -2,25 +2,11 @@
...
-        return Language::_("Digitaloceanmodule.name", true);
-    }
-
-    public function getVersion() {
-        return self::$version;
-    }
-
-    public function getAuthors() {
-        return self::$authors;
+        $this->loadConfig(dirname(__FILE__) . DS . "composer.json");
```
Comment (tysonphillips) [inline digitaloceanmodule.php]: Why are you loading a composer file as the module config file? A Blesta config file (https://docs.blesta.com/display/dev/Module+Configuration) and a composer file are very different.

You need to include a config.json file with the appropriate information as described in the link above, or you need to define the required methods. 

I would typically expect separate pull requests for something like this--one to convert the module to use the JSON config, and another for each specific fix to the module.

I see the module is currently on version "2.2", which I assume is semantically "2.2.0". A change should also increment the version to "2.2.1", since unfortunately, the module defines its own version separate from the package itself.

File: digitaloceanmodule.php [hunk 32]
```diff
@@ -2,25 +2,11 @@
 
 class Digitaloceanmodule extends Module {
 
-    private static $version = "2.2";
-    private static $authors = array(array('name' => "ModulesBakery.com.", 'url' => "http://www.modulesbakery.com"));
```
Comment (tysonphillips) [inline digitaloceanmodule.php:6]: ModulesBakery is the original author, so they should be included in any place authors are listed.

Comment (timnboys) [reply digitaloceanmodule.php:6]: yes they are listed as original author that was moved to config.json I am working on right now.

**Comment by timnboys** [comment]
@tysonphillips @pmphillips all of you have asked is done now.
